includes/validation/class-amp-debug.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Class AMP_Debug
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Debug
+ *
+ * @since 1.4.1
+ */
+final class AMP_Debug {
+
+	/**
+	 * A top-level query var for AMP flags.
+	 *
+	 * @var string
+	 */
+	const AMP_FLAGS_QUERY_VAR = 'amp_flags';
+
+	/**
+	 * A query var to disable post processing.
+	 *
+	 * @var string
+	 */
+	const DISABLE_POST_PROCESSING_QUERY_VAR = 'disable_post_processing';
+
+	/**
+	 * A query var to disable the response cache.
+	 *
+	 * @var string
+	 */
+	const DISABLE_RESPONSE_CACHE_QUERY_VAR = 'disable_response_cache';
+
+	/**
+	 * A query var to prevent a redirect to a non-AMP URL.
+	 *
+	 * @var string
+	 */
+	const PREVENT_REDIRECT_TO_NON_AMP_QUERY_VAR = 'prevent_redirect';
+
+	/**
+	 * A query var to disable tree shaking.
+	 *
+	 * @var string
+	 */
+	const DISABLE_TREE_SHAKING_QUERY_VAR = 'disable_tree_shaking';
+
+	/**
+	 * Query var to reject all validation errors.
+	 *
+	 * @var string
+	 */
+	const REJECT_ALL_VALIDATION_ERRORS_QUERY_VAR = 'reject_all_errors';
+
+	/**
+	 * Query var to accept 'excessive_css' validation errors.
+	 *
+	 * @var string
+	 */
+	const ACCEPT_EXCESSIVE_CSS_ERROR_QUERY_VAR = 'accept_excessive_css';
+
+	/**
+	 * Query var to accept 'excessive_css' validation errors.
+	 *
+	 * @var string
+	 */
+	const DISABLE_AMP_QUERY_VAR = 'disable_amp';
+
+	/**
+	 * Gets whether the flag is present as a query var.
+	 *
+	 * @param string $flag The name of the flag (query var).
+	 * @return bool Whether the flag is present as a query var.
+	 */
+	public static function has_flag( $flag ) {
+		if (
+			isset( $_GET[ self::AMP_FLAGS_QUERY_VAR ][ $flag ] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			&&
+			array_key_exists( $flag, self::get_all_query_vars() )
+			&&
+			AMP_Validation_Manager::has_cap()
+		) {
+			return filter_var( $_GET[ self::AMP_FLAGS_QUERY_VAR ][ $flag ], FILTER_VALIDATE_BOOLEAN ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Using `filter_input_array()` would have been better than having to pass `$_GET` to `filter_var()`, but I couldn't figure out how to test this, without something like:

```php
Monkey\Functions\expect( 'filter_input_array' ) 
...
```

Monkey [isn't available](https://github.com/ampproject/amp-wp/blob/65208999a6e817ff7c74afc31b5e622c971326b0/composer.json#L27) in `composer.json`, though we could propose adding it if was important enough. Also, maybe there's another way to mock `filter_input_array()`.